streaming/top-n-stocks/src/main/java/com/hazelcast/jet/sample/operations/TopNOperation.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sample.operations;
+
+import com.hazelcast.jet.function.DistributedBiFunction;
+import com.hazelcast.jet.function.DistributedBinaryOperator;
+import com.hazelcast.jet.function.DistributedComparator;
+import com.hazelcast.jet.function.DistributedFunction;
+import com.hazelcast.jet.function.DistributedSupplier;
+import com.hazelcast.jet.windowing.WindowOperation;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static com.hazelcast.jet.function.DistributedFunction.identity;
+import static com.hazelcast.jet.stream.impl.StreamUtil.checkSerializable;
+
+public class TopNOperation<T> implements WindowOperation<T, List<T>, List<T>> {
+    private final int n;
+    private final Comparator<? super T> comparator;
+
+    public TopNOperation(int n, DistributedComparator<? super T> comparator) {
+        checkSerializable(comparator, "comparator");
+        this.n = n;
+        this.comparator = comparator;
+    }
+
+    @Nonnull
+    @Override
+    public DistributedSupplier<List<T>> createAccumulatorF() {
+        return () -> new ArrayList<>(n);
+    }
+
+    @Nonnull
+    @Override
+    public DistributedBiFunction<List<T>, T, List<T>> accumulateItemF() {
```

Comment:
While this code works, a `TreeMap` could have solved it with one-liners; and a truly appropriate solution would implement a max-heap with a similar amount of code. We already have an implementation in Hazelcast Hot Restart module; it could be generalized and reused here.